std/container/rbtree.d

```diff
@@ -1714,10 +1714,13 @@ assert(equal(rbt[], [5]));
     Formats the RedBlackTree into a sink function. For more info see
     $(D std.format.formatValue)
     */
-    void toString(scope void delegate(const(char)[]) sink, FormatSpec!char fmt) const {
-        sink("RedBlackTree(");
-        sink.formatValue(this[], fmt);
-        sink(")");
+    static if(is(typeof((){FormatSpec!(char) fmt; formatValue((const(char)[]) {}, ConstRange.init, fmt);})))
+    {
+        void toString(scope void delegate(const(char)[]) sink, FormatSpec!char fmt) const {
```

Comment:
the problem with that delegate signature is that toString can not be @safe because you can't call put(const(char)[]) @system delegate.

One solution would be to make it trusted, by that is bad as well. I think at some point we have to make a choice what is the lesser evil. Disallow @safe toString by default or be @trusted and hope.